manager.go

```diff
@@ -58,12 +61,18 @@ func (sm *spinnerManager) AddSpinner(message string) *Spinner {
 		HasUpdate:     sm.hasUpdate,
 	}
 	spinner := NewSpinner(opts)
+
+	sm.spinnersMu.Lock()
 	sm.spinners = append(sm.spinners, spinner)
+	sm.spinnersMu.Unlock()
+
 	return spinner
 }
 
 // GetSpinners returns the spinners managed by the manager.
 func (sm *spinnerManager) GetSpinners() []*Spinner {
```

Comment:
I'd not considered using this method internally.

It's a fair suggestion and I like it, thank you.